### PR TITLE
CNF-13408: Address sonar cloud quality issues

### PR DESCRIPTION
**Pull request by sakhoury**
This PR addresses the following issues:

- Line length exceeding 120 characters.
- Cognitive complexity greater than 15 (requires refactoring functions).
- Unpredictable resource allocation (specified limits for the manager pod: `cpu`, `memory` and `ephemeral-storage`).
- Use a specific version tag for the `ubi-minimal` controller image.
- Replace IPv4, IPv6 and MAC addresses in source code with those reserved for documentation purposes.

The following new source files have been introduced for linting purposes:
- sonar-project.properties
- .golanci.yml
- .github/workflows/ci-job.yml

Note that there are no functional changes introduced in this PR.

### CODE REVIEW

File: api/v1alpha1/clusterinstance_types.go [hunk 1]
```diff
@@ -366,15 +373,15 @@ type ClusterInstanceStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Conditions",xDescriptors={"urn:alm:descriptor:io.kubernetes.conditions"}
+	// List of conditions pertaining to actions performed on the ClusterInstance resource.
 	// +optional
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
-	// ClusterDeploymentRef is a reference to the ClusterDeployment.
+	// Rference to the associated ClusterDeployment resource.
```
Comment (carbonin) [inline api/v1alpha1/clusterinstance_types.go:380]: Typo here. (`Rference`)

File: api/v1alpha1/clusterinstance_types.go [hunk 2]
```diff
@@ -366,15 +373,15 @@ type ClusterInstanceStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Conditions",xDescriptors={"urn:alm:descriptor:io.kubernetes.conditions"}
```
Comment (carbonin) [inline api/v1alpha1/clusterinstance_types.go:369]: Was this important at all?

File: bundle/manifests/siteconfig.clusterserviceversion.yaml [hunk 3]
```diff
@@ -37,7 +37,7 @@ metadata:
               "tang": [
                 {
                   "thumbprint": "1234567890",
-                  "url": "http://10.0.0.1:7500"
+                  "url": "http://1.0.0.1:7500"
```
Comment (carbonin) [inline bundle/manifests/siteconfig.clusterserviceversion.yaml:40]: Why change these? And in any case I'd expect these examples to pull from the reserved subnets for documentation. https://en.wikipedia.org/wiki/Reserved_IP_addresses

File: bundle/manifests/siteconfig.clusterserviceversion.yaml [hunk 4]
```diff
@@ -509,8 +504,13 @@ spec:
                   initialDelaySeconds: 5
                   periodSeconds: 10
                 resources:
+                  limits:
+                    cpu: "2"
+                    ephemeral-storage: 10Gi
+                    memory: 16Gi
```
Comment (carbonin) [inline bundle/manifests/siteconfig.clusterserviceversion.yaml:510]: These are quite large.
Do we have any idea what the actual limits should be?

File: internal/controller/clusterdeployment_reconciler.go [hunk 5]
```diff
@@ -162,15 +164,20 @@ func isOwnedByClusterInstance(ownerRefs []metav1.OwnerReference) bool {
 	return clusterInstanceOwner(ownerRefs) != ""
 }
 
-func (r *ClusterDeploymentReconciler) getClusterInstance(ctx context.Context, cd *hivev1.ClusterDeployment) (*v1alpha1.ClusterInstance, error) {
+func (r *ClusterDeploymentReconciler) getClusterInstance(
+	ctx context.Context,
+	cd *hivev1.ClusterDeployment,
+) (
+	*v1alpha1.ClusterInstance, error) {
```
Comment (carbonin) [inline internal/controller/clusterdeployment_reconciler.go:171]: I really dislike this formatting. Especially how the parens don't match in the return value.
Maybe move the return value up a line?

File: internal/controller/clusterinstance/validator.go [hunk 6]
```diff
@@ -100,21 +59,49 @@ func Validate(ctx context.Context, c client.Client, clusterInstance *v1alpha1.Cl
...
+	if (clusterInstance.Spec.TemplateRefs == nil) || (len(clusterInstance.Spec.TemplateRefs) < 1) {
+		return fmt.Errorf("missing cluster-level TemplateRefs")
+	}
+	// Verify that the cluster-level TemplateRefs exist
+	for _, templateRef := range clusterInstance.Spec.TemplateRefs {
+		cm := &corev1.ConfigMap{}
+		if err := c.Get(ctx, types.NamespacedName{
+			Name:      templateRef.Name,
+			Namespace: templateRef.Namespace,
+		}, cm); err != nil {
```
Comment (carbonin) [inline internal/controller/clusterinstance/validator.go:98]: I also don't live the formatting here.
Maybe split the `NamespacedName` struct into a variable and one-line the conditional.

File: internal/controller/clusterdeployment_reconciler.go (hunk 5 above)
Comment (sakhoury) [reply internal/controller/clusterdeployment_reconciler.go:171]: +1 looks like an unintentional line-break here. Fixed!

File: internal/controller/clusterinstance/validator.go (hunk 6 above)
Comment (sakhoury) [reply internal/controller/clusterinstance/validator.go:98]: Nice suggestion, changed them all!

File: bundle/manifests/siteconfig.clusterserviceversion.yaml (hunk 4 above)
Comment (sakhoury) [reply bundle/manifests/siteconfig.clusterserviceversion.yaml:510]: Initially there were no limits, which in theory implies that the pod could consume much more resources. I put these values as max values. We can revisit them once we have an analysis from the Perf team? Feel free to suggest alternative limits :)

File: bundle/manifests/siteconfig.clusterserviceversion.yaml (hunk 3 above)
Comment (sakhoury) [reply bundle/manifests/siteconfig.clusterserviceversion.yaml:40]: I meant to change to loopback address! Updated.

File: api/v1alpha1/clusterinstance_types.go (hunk 2 above)
Comment (sakhoury) [reply api/v1alpha1/clusterinstance_types.go:369]: Nope, got it from somewhere else. Cleaned it up now!

File: api/v1alpha1/clusterinstance_types.go (hunk 1 above)
Comment (sakhoury) [reply api/v1alpha1/clusterinstance_types.go:380]: Good catch! Fixed!

File: bundle/manifests/siteconfig.clusterserviceversion.yaml [hunk 7]
```diff
@@ -85,15 +85,15 @@ metadata:
                         "ipv4": {
                           "address": [
                             {
-                              "ip": "10.16.231.3",
+                              "ip": "1.2.3.4",
```
Comment (carbonin) [inline bundle/manifests/siteconfig.clusterserviceversion.yaml:88]: I mean even these should probably be using the reserved IP range for docs/examples.
This comment applies to all IP addresses in this example text
https://en.wikipedia.org/wiki/Reserved_IP_addresses

Comment (sakhoury) [reply bundle/manifests/siteconfig.clusterserviceversion.yaml]: Great suggestion! Have updated the IP{v4,v6} addresses and MAC addresses as well!

**Review by carbonin — APPROVED** [review]

**Pull request merged**